app/src/main/java/org/simple/clinic/navigation/v2/Router.kt

```diff
@@ -426,6 +422,14 @@ class Router(
       throw RuntimeException("Can only execute navigation state changes on the UI thread! Current thread is [${Thread.currentThread().name}].")
     }
   }
+
+  private fun isModalBeforeLastFullScreen(
```

Comment:
While the use case of this method currently is only in a method which deals with modal top keys, the implementation of this method itself has nothing to do with modals and only checks if a given nav request is before another. This also exposes the hidden implementation detail that `History` maintains the nav requests as a list when the expectation is that it is a stack.

Can we instead rename this method to better reflect this and then make it a part of the `NavRequest` type so that it can be more readable?

Ex:
```
|| history.isNavRequestBeforeAnother(navRequest, lastFullScreen)
```